pkg/extensions/sync/signatures.go

```diff
@@ -448,6 +354,61 @@ func (sig *signaturesCopier) syncOCIRefs(localRepo, remoteRepo, digestStr string
 	return nil
 }
 
+func (sig *signaturesCopier) syncOCIArtifact(localRepo, remoteRepo, reference string,
+	ociArtifactBuf []byte,
+) error {
+	var ociArtifact ispec.Artifact
+
+	err := json.Unmarshal(ociArtifactBuf, &ociArtifact)
+	if err != nil {
+		sig.log.Error().Err(err).Msgf("couldn't unmarshal OCI artifact from %s:%s", remoteRepo, reference)
+
+		return err
+	}
+
+	canSkipOCIArtifact, err := sig.canSkipOCIArtifact(localRepo, reference, ociArtifact)
+	if err != nil {
+		sig.log.Error().Err(err).Msgf("couldn't check if OCI artifact %s:%s can be skipped",
+			remoteRepo, reference)
+	}
+
+	if canSkipOCIArtifact {
+		return nil
+	}
+
+	imageStore := sig.storeController.GetImageStore(localRepo)
+
+	sig.log.Info().Msg("syncing OCI artifacts")
+
+	for _, blob := range ociArtifact.Blobs {
+		if err := syncBlob(sig, imageStore, localRepo, remoteRepo, blob.Digest); err != nil {
+			return err
+		}
+	}
+
+	artifactManifestBuf, err := json.Marshal(ociArtifact)
```

Comment:
I think this is my mistake, we already have ociArtifactBuf, no need to marshal again :D.